test/components/autofillTest.js

```diff
@@ -256,6 +256,174 @@ describe('Autofill', function () {
+        .getValue('[name="15_country"]').should.eventually.be.equal('United States')
+    })
+    it('Deleting Address', function * () {
+      yield this.app.client
+        .tabByIndex(0)
+        .loadUrl(page1Url)
+        .url(getTargetAboutUrl(page1Url))
+        .waitForVisible('[title="Delete address"]')
+        .click('[title="Delete address"]')
+        .waitForVisible('[data-l10n-id=noAddressesSaved]')
+    })
+    const addCreditCardButton = '.addCreditCardButton'
+    const saveCreditCardButton = '.saveCreditCardButton'
+    const cardName = 'Test Card'
+    const cardNumber = '1234567890'
+    const expMonth = 10
+    const expYear = new Date().getFullYear() + 2
+    it('Adding Credit Card', function * () {
+      yield this.app.client
+        .tabByIndex(0)
+        .loadUrl(page1Url)
+        .url(getTargetAboutUrl(page1Url))
+        .waitForVisible(addCreditCardButton)
+        .click(addCreditCardButton)
+        .windowByUrl(Brave.browserWindowUrl)
+        .waitForVisible(autofillCreditCardPanel)
+        .click('#nameOnCard')
+        .keys(cardName)
+        .click('#creditCardNumber')
+        .keys(cardNumber)
+        .click(saveCreditCardButton)
+        .waitUntil(function () {
+          return this.getAppState().then((val) => {
+            return val.value.autofill.creditCards.length === 1
+          })
+        })
+        .tabByUrl(this.page1Url)
+        .waitForVisible('.autofillPage')
+        .getText('.creditCardName').should.eventually.be.equal(cardName)
+        .getText('.creditCardNumber').should.eventually.be.equal('***' + cardNumber.slice(-4))
+        .getText('.creditCardPExpirationDate').should.eventually.be.equal('01/' + new Date().getFullYear().toString())
+    })
+    it('Editing Credit Card', function * () {
+      yield this.app.client
+        .tabByIndex(0)
+        .loadUrl(page1Url)
+        .url(getTargetAboutUrl(page1Url))
+        .waitForVisible('[title="Edit creditCard"]')
+        .click('[title="Edit creditCard"]')
+        .windowByUrl(Brave.browserWindowUrl)
+        .waitForVisible(autofillCreditCardPanel)
+        .selectByValue('.expMonthSelect', expMonth.toString())
+        .selectByValue('.expYearSelect', expYear.toString())
+        .click(saveCreditCardButton)
+        .waitUntil(function () {
+          return this.getAppState().then((val) => {
+            return val.value.autofill.creditCards.length === 1
+          })
+        })
+        .tabByUrl(this.page1Url)
+        .refresh()
+        .waitForVisible('.autofillPage')
+        .getText('.creditCardName').should.eventually.be.equal(cardName)
+        .getText('.creditCardNumber').should.eventually.be.equal('***' + cardNumber.slice(-4))
+        .getText('.creditCardPExpirationDate').should.eventually.be.equal(
+          expMonth.toString() + '/' + expYear.toString())
+    })
+    it('Credit Card form test', function * () {
+      const page1Url = 'https://www.roboform.com/filling-test-all-fields'
```

Comment:
Can this be stubbed / replaced with a fixture? (or maybe it is and I just didn't see it).  Would be ideal if we didn't have a dependency on their site :smile: